### PR TITLE
Fixes related to handling of second qualification forms

### PR DESCRIPTION
**Pull request by harel**
## Description of change

use an isoformated date for as search. 
check if emt was provided in second qual form. 
use just date when filtering for second qual to catch discrepancies between log date and 2nd qual submission

### CODE REVIEW

File: app/enquiries/common/email_campaign_utils.py [hunk 1]
```diff
@@ -179,6 +184,13 @@ def process_enquiry_update(emt_id, phone=None, consent=None):
     client = AdobeClient()
     try:
         enquiry = Enquiry.objects.get(id=emt_id)
+        # check if this has already been processed
+        has_processed = EnquiryActionLog.objects.filter(
+            enquiry=enquiry,
+            action=ref_data.EnquiryAction.SECOND_QUALIFICATION_FORM
+        ).exists()
+        if has_processed:
+            return log
```
Comment (mforner13) [inline app/enquiries/common/email_campaign_utils.py:193]: What's the purpose of returning `None` here and on line 177? If the return value doesn't matter can we just `return`?

Comment (harel) [reply app/enquiries/common/email_campaign_utils.py:193]: It's just a habit. In a happy path it returns the log object , which is useful if you're testing or debugging. So as a habit if i return something, i try to return None explicitly if there is a None value to return . 
I can change to an empty return if that works better for you though. i don't mind

Comment (peterhudec) [reply app/enquiries/common/email_campaign_utils.py:193]: The return value doesn't seem to be used apart from a test, where `None` is asserted, so the `log` variable can be removed.

File: app/enquiries/tests/test_adobe_integration.py [hunk 2]
```diff
@@ -182,3 +182,8 @@ def test_log_action(self):
 
         last_action_date = EnquiryActionLog.get_last_action_date(action)
         self.assertEqual(last_action_date.enquiry.id, self.enquiry.id)
+
+    @freeze_time()
+    def test_process_enquiry_update_bad_id(self):
+        log = campaign.process_enquiry_update(None)
+        assert log is None
```
Comment (peterhudec) [inline app/enquiries/tests/test_adobe_integration.py:189]: This kind of tests are of little value as it only checks one specific scenario. I would remove it.

File: app/enquiries/common/email_campaign_utils.py [hunk 3]
```diff
@@ -170,6 +172,9 @@ def process_enquiry_update(emt_id, phone=None, consent=None):
     and new stage, to update the relevant profile.
     """
     log = None
+    if not emt_id:
+        logger.warning("Emt_id not provided for second qualification form")
+        return log
```
Comment (peterhudec) [inline app/enquiries/common/email_campaign_utils.py:177]: This check should be done outside of the function and the function should only be called if `emt_id` is not `None`. The `emt_id` is after all defined as a _required_ parameter.

**Review by peterhudec — CHANGES_REQUESTED** [review]
Looks good, just a couple of things.

File: app/enquiries/tests/test_adobe_integration.py (hunk 2 above)
Comment (harel) [reply app/enquiries/tests/test_adobe_integration.py:189]: i only tried to get it in because coverage was complaining.

Comment (mforner13) [reply app/enquiries/tests/test_adobe_integration.py:189]: As codecov passing is not mandatory for the PR to be merged, I'd say best to remove this if it's only to placate the check, so as not to give false reassurance?

**Review by mforner13 — APPROVED** [review]

**Review by peterhudec — APPROVED** [review]

**Pull request merged**